db/migrate/20150202094307_add_range_to_trend_counters.rb

```diff
@@ -0,0 +1,22 @@
+class AddRangeToTrendCounters < ActiveRecord::Migration
+  def change
+    add_column :trend_counters, :interval_start, :datetime
+    add_column :trend_counters, :interval_end, :datetime
+
+    trend_lines = TrendCounter.order(:trend_id).group_by {|counter| counter.trend_id}
+
+    trend_lines.each do |trend_id, line|
+      sorted_counters = line.sort_by {|counter| counter.created_at}
+
+      first_record = sorted_counters.first
+      first_record.interval_start = first_record.created_at
+
+      sorted_counters.drop(1).each_with_index do |counter, idx|
```

Comment:
The proper word would be skip. It sets the `#end_interval` of the previous counter to the start of the next one: `a[i].end = a[i+1].start`. I am updating sorted_counters (unshifted) with values from `#each_with_index` (shifted)